### DISCUSSION TITLE
Plugins for extensions

### DISCUSSION THREAD

**Discussion by henryiii** [python-poetry/poetry#3857] [Ideas]
I'd like to use the plugin system to enable extension building plugins. I'm working with scikit-build, the official build system for CMake in Python, and I'd like to (over the next two years) build a plugin for Poetry that would allow the integration of CMake builds with Poetry projects. The idea being you could include this hypothetical `scikit-build-poetry` alongside poetry in your pyproject.toml (or any other builder plugin, like a setuptools based plugin or enscons based plugin, but this is the one I'd be interested in working on), and then specify extensions to build with it (settings in `[tool.scikit-build-poetry]` would be fine, or there could be a built in way to specify extension locations).

Based on https://github.com/python-poetry/poetry/pull/3733#issuecomment-809755209, this is basically already almost possible, but not very well integrated. What do you think? Also, might there need to be a bit more added to enable high quality integration with extension building plugins, or is #3733 enough?

**Comment by abn** [comment]
@henryiii my recommendation is to try it, and raise issues or features as you encounter a need for it. The pluging system is in purpose limited at the moment as it is quite early. However, going forward we would definitely expand scope for more tighter integrations. But until the community starts developing against it and point out shortcomings, we cannot do much else than guess. :)

**Reply by cwittlinger** [reply]
Is the plugin system already available in 1.1.6? I feel the plugin commands are only in master, but not yet available otherwise?

**Reply by abn** [reply]
Correct. The first release to contain it will be `1.2.0a0`.
